binding/nodejs/src/cobra.ts

```diff
@@ -0,0 +1,181 @@
+        `File not found at 'libraryPath': ${libraryPath}`
+      );
+    }
+
+    const pvCobra = require(libraryPath); // eslint-disable-line
+    this._pvCobra = pvCobra;
+
+    let cobraHandleAndStatus: CobraHandleAndStatus | null = null;
+    try {
+      pvCobra.set_sdk('nodejs');
+
+      cobraHandleAndStatus = pvCobra.init(accessKey);
+    } catch (err: any) {
+      pvStatusToException(PvStatus[err.code as keyof typeof PvStatus], err);
+    }
+
+    const status = cobraHandleAndStatus!.status;
+    if (status !== PvStatus.SUCCESS) {
+      this.handlePvStatus(status, 'Cobra failed to initialize');
+    }
+
+    this._handle = cobraHandleAndStatus!.handle;
+    this._sampleRate = pvCobra.sample_rate();
+    this._frameLength = pvCobra.frame_length();
+    this._version = pvCobra.version();
+  }
+
+  /**
+   * @returns number of audio samples per frame (i.e. the length of the array provided to the process function)
+   * @see {@link process}
+   */
+  get frameLength(): number {
+    return this._frameLength;
+  }
+
+  /**
+   * @returns the audio sampling rate accepted by the process function
+   * @see {@link process}
+   */
+  get sampleRate(): number {
+    return this._sampleRate;
+  }
+
+  /**
+   * @returns the version of the Cobra engine
+   */
+  get version(): string {
+    return this._version;
+  }
+
+  /**
+   * Processes a frame of audio and returns the voice probability result.
+   *
+   * @param {Int16Array} pcm Audio data. The audio needs to have a sample rate equal to `Cobra.sampleRate` and be 16-bit linearly-encoded.
+   * The specific array length can be attained by calling `Cobra.frameLength`. This function operates on single-channel audio.
+   * @returns {number} Voice probability result.
+   */
+  process(pcm: Int16Array): number {
+    assert(pcm instanceof Int16Array);
+
+    if (
+      this._handle === 0 ||
+      this._handle === null ||
+      this._handle === undefined
+    ) {
+      throw new CobraInvalidStateError('Cobra is not initialized');
+    }
+
+    if (pcm === undefined || pcm === null) {
+      throw new CobraInvalidArgumentError(
+        `PCM array provided to 'Cobra.process()' is undefined or null`
+      );
+    } else if (pcm.length !== this.frameLength) {
+      throw new CobraInvalidArgumentError(
+        `Size of frame array provided to 'Cobra.process()' (${pcm.length}) does not match the engine 'Cobra.frameLength' (${this.frameLength})`
+      );
+    }
+
+    let VoiceProbabilityAndStatus: VoiceProbabilityAndStatus | null = null;
+    try {
+      VoiceProbabilityAndStatus = this._pvCobra.process(this._handle, pcm);
+    } catch (err: any) {
+      pvStatusToException(PvStatus[err.code as keyof typeof PvStatus], err);
+    }
+
+    const status = VoiceProbabilityAndStatus!.status;
+    if (status !== PvStatus.SUCCESS) {
+      this.handlePvStatus(status, 'Cobra failed to process the audio frame');
+    }
+
+    return VoiceProbabilityAndStatus!.is_voiced;
```

Comment:
Weird capitalization here
```suggestion
    let voiceProbabilityAndStatus: VoiceProbabilityAndStatus | null = null;
    try {
      voiceProbabilityAndStatus = this._pvCobra.process(this._handle, pcm);
    } catch (err: any) {
      pvStatusToException(PvStatus[err.code as keyof typeof PvStatus], err);
    }

    const status = voiceProbabilityAndStatus!.status;
    if (status !== PvStatus.SUCCESS) {
      this.handlePvStatus(status, 'Cobra failed to process the audio frame');
    }

    return voiceProbabilityAndStatus!.is_voiced;
```